File: src/main/java/de/danielmaile/aether/listeners/ListenerWorldSave.java
package de.danielmaile.aether.listeners;

import de.danielmaile.aether.worldgen.AetherWorld;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.world.WorldSaveEvent;

public class ListenerWorldSave implements Listener
{
    @EventHandler
    public void onSave(WorldSaveEvent event)
    {
        if(!event.getWorld().equals(AetherWorld.getWorld())) return;
        AetherWorld.getObjectManager().save();
    }
}

File: src/main/java/de/danielmaile/aether/listeners/ListenerWorldSave.kt
package de.danielmaile.aether.listeners

import de.danielmaile.aether.inst
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.world.WorldSaveEvent

class ListenerWorldSave : Listener {

    @EventHandler
    fun onSave(event: WorldSaveEvent) {
        if (event.world !=  inst().aetherWorld.getWorld()) return
        inst().getObjectManager().save()
    }
}